File: FORK/fork_4.c

#include<stdio.h>
#include<unistd.h>
#include<sys/types.h>
#include<sys/wait.h>

int main()
{
 int p;
 int a=2;
 int t=getpid();
 a++;
 fork();
 
 printf("\nValue is %d , address is %p , PID is %d and PPID is %d\n",a,&a,getpid(),getppid());
 
 wait(&p);
 
 return 0;
}
 

File: OS MODULE PYTHON/fork4.py
import os

pid = os.fork()

if(pid>0):
  print("Hello I'm parent process")
  print("Value returned by fork is ",pid)
  print("PID is ",os.getpid(),"PPID is ",os.getppid())
  
else:
  print("Hello I'm child process")
  print("Value returned by fork is ",pid)
  print("PID is ",os.getpid(),"PPID is ",os.getppid())
  

